src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1402 @@
+
+        hidden_state = self.add_position_embeddings(hidden_state, text_end_positions, multiway_split_position)
+        hidden_state = hidden_state * (1 - encoder_padding_mask.unsqueeze(-1).type_as(hidden_state))
+
+        # past_key_value is not None during inference if we use the bidirectional encoder as a generator as in s2s-ft (https://arxiv.org/abs/2110.13640)
+        for idx, layer in enumerate(self.layers):
+            if output_hidden_states:
+                all_hidden_states = all_hidden_states + (hidden_state,)
+
+            hidden_state = layer(
+                hidden_state,
+                encoder_padding_mask=encoder_padding_mask if past_key_value is None else None,
+                attention_mask=attention_mask,
+                multiway_split_position=multiway_split_position,
+                past_key_value=past_key_value[idx] if past_key_value is not None else None,
+                output_attentions=output_attentions,
+            )
+
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (hidden_state,)
+
+        if self.fc_norm is not None:
+            hidden_state = self.fc_norm(hidden_state, split_position=multiway_split_position)
+
+        if not return_dict:
+            return tuple(
+                v
+                for v in [
+                    hidden_state,
+                    all_hidden_states,
+                    all_self_attentions,
+                ]
+                if v is not None
+            )
+
+        return BaseModelOutput(
+            last_hidden_state=hidden_state,
+            hidden_states=all_hidden_states,
+            attentions=all_self_attentions,
+        )
+
+
+@add_start_docstrings(
+    """BEiT-3 is a general-purpose multimodal foundation model that excels in both vision and vision-language tasks. It
+        utilizes [Multiway transformers] (https://arxiv.org/abs/2208.10442) for deep fusion and modality-specific
+        encoding, and unifies masked modeling on images, texts, and image-text pairs, achieving top performance on
+        multiple benchmarks.""",
+    BEIT3_START_DOCSTRING,
+)
+class Beit3Model(Beit3PreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        self.text_embedding = nn.Embedding(config.vocab_size, config.hidden_size)
+        self.vision_embedding = Beit3VisionEmbedding(config)
+        self.encoder = Beit3Encoder(config)
+        self.post_init()
+
+    def get_input_embeddings(self):
+        return self.text_embedding
+
+    def set_input_embeddings(self, value: nn.Embedding) -> None:
+        self.text_embedding = value
+
+    def get_num_layers(self):
+        return self.encoder.num_layers
+
+    @add_start_docstrings_to_model_forward(BEIT3_MODEL)
+    def forward(
+        self,
+        input_ids=None,
+        pixel_values=None,
+        text_padding_mask=None,
+        attention_mask=None,
```

Comment:
would be great to only have attention_mask here, since the preprocessor class (Beit3Processor) does not prepare the text_padding_mask.

Ideally one should be able to do:
```
from transformers import Beit3Processor, Beit3Model

processor = Beit3Processor.from_pretrained("...")
model = Beit3Model.from_pretrained("...")

inputs = processor(text=text, images=images, return_tensors="pt")

outputs = model(**inputs)
```
i.e. just unroll the dictionary created by the preprocessor